replications/internal/queue_management.go

```diff
@@ -136,62 +136,51 @@ func (rq *replicationQueue) run() {
 			// that rq.SendWrite will be called again in this situation and not leave data in the queue. Outside of this
 			// specific scenario, the buffer might result in an extra call to rq.SendWrite that will immediately return on
 			// EOF.
-			for rq.SendWrite() {
-			}
+			rq.SendWrite()
```

Comment:
After some refactoring associated with the scanner advancement, it came to the point where this function would only ever return `false`, so looping over it is no longer needed.

Reply:
Turns out that there is a reason to continue to loop over this - when testing with small segment sizes so that multiple segments would be written through, there are situations where `Next()` returns falls even if there is more data in the queue. So to take care of that I modified the test and restored the looping / bool returning behavior.